src/models/devicemodel.cpp

```diff
@@ -71,20 +71,21 @@ bool DeviceModel::fromSettings(const Keys* keys) {
 
 namespace {
 
-bool sortCallback(const Device& a, const Device& b, const Keys* keys) {
-  if (a.isCurrentDevice(keys)) {
-    return true;
-  }
-
-  if (b.isCurrentDevice(keys)) {
-    return false;
-  }
-
```

Comment:
Why remove this, out of curiousity?

Reply:
There is a static assert in `std::sort` that requires the sort comparator to satisfy strict-weak ordering. It is implemented by invoking, `compare(a, a)` and `compare(b, b)` both of which must always evaluate to false. When compiled for debug we hit this assert and the app crashes.

I am not sure if this is just a simulator thing, or an Xcode 16 thing, but I figured it was better to fix it.

Reply:
The fix was implemented by sorting on creation time, and then add a second step which looks for the current device and moves it to the head of the list.